src/lib/core/selection/selection.ts

```diff
@@ -0,0 +1,168 @@
+import {Subject} from 'rxjs/Subject';
+
+
+/**
+ * Class to be used to power selecting one or more options from a list.
+ * @docs-private
+ */
+export class MdSelectionModel {
+  constructor(
+    private _options: any[],
+    private _isMulti = false,
+    initiallySelectedValues?: any[]) {
+
+    if (initiallySelectedValues && initiallySelectedValues.length) {
+      if (_isMulti) {
+        initiallySelectedValues.forEach(value => this._select(value));
+      } else {
+        this._select(initiallySelectedValues[0]);
+      }
+
+      // Clear the array in order to avoid firing the change event for preselected values.
+      this._unflushedSelectedValues.length = 0;
+    }
+  }
+
+  /** Event emitted when the value has changed. */
+  onChange: Subject<MdSelectionChange> = new Subject();
+
+  /** Currently-selected values. */
+  private _selectedValues: any[] = [];
+
+  /** Keeps track of the deselected options that haven't been emitted by the change event. */
+  private _unflushedDeselectedValues: any[] = [];
```

Comment:
Not sure "unflushed" is the right term here. How about something like "pendingDeselection", "deselectedToEmit"?